File: app/src/main/java/com/dashingqi/dqkotlin/sealed/Person.java
package com.dashingqi.dqkotlin.sealed;

/**
 * 定义一个枚举类
 * @author zhangqi61
 * @since 2021/12/27
 */
public enum Person {
    CHILDRNE("efdf"),
    MAN(1),
    WOMEN(2);
    private int age;
    private String name;

    private Person(int age) {
        this.age = age;
    }

    private Person(int age, String name) {
        this.age = age;
        this.name = name;
    }

    public int getAge() {
        return age;
    }

    public void setAge(int age) {
        this.age = age;
    }
}

File: app/src/main/java/com/dashingqi/dqkotlin/extend/Person.kt
package com.dashingqi.dqkotlin.extend

/**
 * @author zhangqi61
 * @since 2022/1/21
 */
open class Person(private val age: Int, var name: String) {
    //    fun isAdult() = age > 18
    // 将isAdult 变成Person的属性
    val isAdult
        get() = age > 19

    // 该情况下自动类型推导会失效，需要声明一下类型
    val isMe: Boolean
        get() {
            print("")
            return age > 18
        }

    // 这种代码无法正常工作
    val isAdult2 = age > 19

    var length: Int = 180
        // 这是length属性的setter的方法
        // private 表示限制在类内部访问
        private set(value: Int) {
            print("")
            field = value
            print("")
        }

    // 声明成open 才能被子类重写
    open fun doSomething() {

    }
}